bob/src/cluster/quorum.rs

```diff
@@ -351,25 +352,194 @@ impl Cluster for Quorum {
     }
 
     async fn exist(&self, keys: &[BobKey]) -> Result<Vec<bool>, Error> {
-        let keys_by_nodes = group_keys_by_nodes(&self.mapper, keys);
-        debug!(
-            "EXIST Nodes for fan out: {:?}",
-            &keys_by_nodes.keys().flatten().collect::<Vec<_>>()
-        );
         let len = keys.len();
-        let mut exist = vec![false; len];
-        for (nodes, (keys, indexes)) in keys_by_nodes {
-            let cluster_results = LinkManager::exist_on_nodes(&nodes, &keys).await;
-            for result in cluster_results.into_iter().flatten() {
-                for (&r, &ind) in result.inner().iter().zip(&indexes) {
-                    exist[ind] |= r;
+        debug!("EXIST {} keys", len);
+
+        let mut result = vec![false; len];
+
+        let (local, mut primary, mut secondary) = group_by_nodes(keys, &self.mapper);
+
+        if let Some(local) = local {
+            if !local.is_empty() {
+                let local_keys = local.collect(keys);
+                let local_exist = exist_on_local_node(&self.backend, &local_keys).await?;
```

Comment:
It is better to use the same error processing approach as it is implemented in `get`: do not stop processing on the error here. If error happened, then it should be written to the log on `warn!` level and processing should continue

Reply:
Please add logs so that the progress of the function can be tracked. 
Add log describing `exist_on_local_node`: on error log that error, on success log on `trace` level the number of keys passed and the number of keys presented locally